### PR TITLE
CallCompressed and compression_type 

### PR DESCRIPTION
**Pull request by joshblum**
Adds support for new RPC call `CallCompressed` and a `@compression_type` decorator. A protocol can specify a top level `@compression_type` (defaults to `none`) and individual methods can override this default.

depends on changes in https://github.com/keybase/go-framed-msgpack-rpc/pull/162

### CODE REVIEW

File: src/emit.iced [hunk 1]
```diff
@@ -628,7 +637,7 @@ exports.GoEmitter = class GoEmitter
     @output "},"
 
   emit_server_hook_method_type : ({name, details}) ->
-    @output "MethodType: rpc.Method#{if is_one_way(details) then 'Notify' else 'Call'},"
+    @output "MethodTypes: []rpc.MethodType{#{if is_one_way(details) then 'rpc.MethodNotify' else 'rpc.MethodCall, rpc.MethodCallCompressed'}},"
```
Comment (joshblum) [inline src/emit.iced]: i'm pretty sure this is entirely unused, should we just kill it?

File: src/emit.iced [hunk 2]
```diff
@@ -7,7 +7,7 @@ pkg = require '../package.json'
 
 #====================================================================
 
-is_one_way = (d) -> (d.notify? or d.oneway)
+is_one_way = (d) -> (d.notify? or d.oneway?)
```
Comment (maxtaco) [inline src/emit.iced]: i challenge this.

File: src/emit.iced (hunk 1 above)
Comment (maxtaco) [reply src/emit.iced]: ok, if unused, yeah...

**Comment by joshblum** [comment]
waiting on https://github.com/keybase/go-framed-msgpack-rpc/pull/162 and manual client testing before merge

**Comment by joshblum** [comment]
depends on https://github.com/keybase/node-avdl-compiler/pull/23

**Pull request merged**